buffer/src/main/java/io/netty/buffer/PoolChunkList.java

```diff
@@ -190,37 +192,41 @@ private static int minUsage0(int value) {
 
     @Override
     public Iterator<PoolChunkMetric> iterator() {
-        if (head == null) {
-            return EMPTY_METRICS;
-        }
-        List<PoolChunkMetric> metrics = new ArrayList<PoolChunkMetric>();
-        for (PoolChunk<T> cur = head;;) {
-            metrics.add(cur);
-            cur = cur.next;
-            if (cur == null) {
-                break;
+        synchronized (arena) {
+            if (head == null) {
+                return EMPTY_METRICS;
+            }
+            List<PoolChunkMetric> metrics = new ArrayList<PoolChunkMetric>();
+            for (PoolChunk<T> cur = head;;) {
+                metrics.add(cur);
+                cur = cur.next;
+                if (cur == null) {
+                    break;
+                }
             }
+            return metrics.iterator();
         }
-        return metrics.iterator();
     }
 
     @Override
     public String toString() {
-        if (head == null) {
-            return "none";
-        }
+        synchronized (arena) {
+            if (head == null) {
+                return "none";
+            }
 
-        StringBuilder buf = new StringBuilder();
-        for (PoolChunk<T> cur = head;;) {
-            buf.append(cur);
-            cur = cur.next;
-            if (cur == null) {
-                break;
+            StringBuilder buf = new StringBuilder();
+            for (PoolChunk<T> cur = head;;) {
+                buf.append(cur);
```

Comment:
consider building the string outside the synchronized block. Trade off between potentially inconsistent state in the string and reducing the critical area.